go/tasks/plugins/array/awsbatch/monitor.go

```diff
@@ -132,16 +137,20 @@ func CheckSubTasksState(ctx context.Context, taskMeta core.TaskExecutionMetadata
 		errorMsg := msg.Summary(cfg.MaxErrorStringLength)
 		parentState = parentState.SetReason(errorMsg)
 	}
+	_, version := currentState.GetPhase()
 	if phase == arrayCore.PhaseCheckingSubTaskExecutions {
-		newPhaseVersion := uint32(0)
-		// For now, the only changes to PhaseVersion and PreviousSummary occur for running array jobs.
-		for phase, count := range parentState.GetArrayStatus().Summary {
-			newPhaseVersion += uint32(phase) * uint32(count)
+		newSubTaskPhaseHash, err := parentState.GetArrayStatus().HashCode()
```

Comment:
Does that take into account the retries? or just the latest attempt?

Reply:
So if there is a subtask failure the phase for that subtask will change to RetryableFailure within this iteration and thus the newSubTaskPhaseHash != currentSubTaskPhaseHash. On the next iteration we will start the next retry pod (which updates the subtask phase to Queued or Running (forget which) and there will be another phaseVersion increase. This should be the desired behavior because the overall map task phase will be reported as Running and the phaseVersion increments will ensure that flyteAdmin processes them.

I think that this answers your question. We are just checking if the phase changes and for retries we make sure to log a retryable failure before transitioning to running again.

Reply:
ah I missed the fact that we don't use the hash directly, we just bump the version... I was worried that the hash can circle back and be a) a previously seen value (which will happen but it's ok given the logic here) or b) be less than the current, which is not ok but doesn't matter given you only compare to previous...